src/middlewares/createTelemetryMiddleware.ts

```diff
@@ -0,0 +1,81 @@
+import { Middleware } from '../types';
+
+type TelemetryWidget = {
+  type: string;
+  params: string[];
+  officialWidget: boolean;
+};
+
+export function createTelemetryMiddleware(): Middleware {
+  const isTelemetryEnabled =
+    typeof window !== 'undefined' &&
+    window.navigator.userAgent.indexOf('Algolia Crawler') > -1;
+
+  if (!isTelemetryEnabled) {
+    return () => ({
+      onStateChange() {},
+      subscribe() {},
+      unsubscribe() {},
+    });
+  }
+
+  const payload: { widgets: TelemetryWidget[] } = {
+    widgets: [],
+  };
+
+  const payloadContainer = document.createElement('meta');
+  const refNode = document.querySelector('head')!;
+  payloadContainer.name = 'instantsearch:widgets';
+  refNode.appendChild(payloadContainer);
+
+  return ({ instantSearchInstance }) => {
+    return {
+      onStateChange() {},
+      subscribe() {
+        setTimeout(() => {
```

Comment:
I don't remember correctly. Could you explain again about the Intermittent issue? IIRC, it was something about `getScopedResults` not having any result yet sometimes. Is `setTimeout` still the valid way to ensure it? I think you said you were going to fix it but not sure if it's applied here or not.

Reply:
it was about the `state` coming from the `results` at the time, but I changed it to come from the helper instead (like the index widget); and that works as expected!

Reply:
I'm not sure to understand. Can you explain it with a comment if appropriate? (Otherwise, explain it here with more context?)

Reply:
at an earlier time the widgets were getWidgetRenderState'ed using "results.getState", but that would fail if the results aren't yet available. Now I changed it to only request getWidgetRenderState using info that's given at init,and thus can't fail anymore @francoischalifour